providers/salesforceMarketing.go

```diff
@@ -0,0 +1,44 @@
+package providers
+
+const SalesforceMarketing Provider = "salesforceMarketing"
+
+func init() {
+	// SalesforceMarketing configuration
+	SetInfo(SalesforceMarketing, ProviderInfo{
+		DisplayName: "Salesforce Marketing Cloud",
+		AuthType:    Oauth2,
+		BaseURL:     "https://{{.workspace}}.rest.marketingcloudapis.com",
+		Oauth2Opts: &Oauth2Opts{
+			GrantType:                 AuthorizationCode, // There is also a s2s clientcredentials option.
+			AuthURL:                   "https://{{.workspace}}.auth.marketingcloudapis.com/v2/authorize",
+			TokenURL:                  "https://{{.workspace}}.auth.marketingcloudapis.com/v2/token",
+			ExplicitScopesRequired:    true,
```

Comment:
According to https://developer.salesforce.com/docs/marketing/marketing-cloud/guide/access-token-s2s.html, "If you don’t include the scope parameter in the request, the token is issued with the scopes specified on the API integration in Installed Packages.". That's way more convenient for our builders to just default to the set of scopes they create the API integration with (on the Salesforce dashboard), so please modify this to "ExplicitScopesRequired: false"

Also I couldn't find any Slab docs published for marketing cloud, did you already create one?

Reply:
Thanks, Read [this](https://developer.salesforce.com/docs/marketing/marketing-cloud/guide/access-token-app.html) and made me think offline is necessary in the request thus scope was necessary, I have just tested without it though and it works just fine.  Updating this.

<img width="573" alt="Screenshot 2024-08-17 at 12 51 23" src="https://github.com/user-attachments/assets/c8e995d0-eddb-4b38-93f3-537ac20ff179">

I have just published the slab now.